ClinGen/clincoded
# Connects to #867 (progress bar), Connects to #719 (pathogenicity calculator), Connects to #849 (missing BA1)

This PR is related to issues adding pathogenicity calculator (ticket #719), displaying calculated result in a progress bar and relocating criteria bar (ticket #867) and adding BA1 into criteria bar (ticket #849).

**Code Changes**

1. Created a new file `src/clincoded/static/components/variant_central/interpretation/shared/calculator.js`. It calculates criteria values in an interpretation and generates a progress bar to display calculated results.
2. In file `src/clincoded/static/components/variant_central/interpretation.js`, added code to display  the progress bar.
3. Added BA1 into criteria bar in file `src/clincoded/static/components/variant_central/interpretation/criteria.js`.
4. Changed the width for each criteria and added new style classes for progress bar in file `src/clincoded/static/scss/clincoded/modules/_curator.scss`
5. Tag `<CurationInterpretationCriteria interpretation={interpretation} />` is removed from file `src/clincoded/static/components/variant_central/interpretation.js` and added in file `src/clincoded/static/components/variant_central/index.js` to relocate criteria bar.

**Test**
Right now, test is limited at 3 criterion in Population tab only.
* Go to VCI, select ClinVar Variation ID, enter 54321 and click Save.
* Click Start New Interpretation, expect to see:
![screen shot 2016-08-10 at 12 57 39 pm](https://cloud.githubusercontent.com/assets/9206696/17568785/1030240a-5efa-11e6-81a1-b856fa58b7c1.png)
* Go to Population tab, select BA1 to Met and click Save. Expect to see:
![screen shot 2016-08-10 at 12 59 47 pm](https://cloud.githubusercontent.com/assets/9206696/17568839/6206584e-5efa-11e6-9e09-3599f94ad303.png)
* Change BA1 to Not Evaluated, select PM2 to Very Strong and click Save. Expect to see:
![screen shot 2016-08-10 at 1 02 48 pm](https://cloud.githubusercontent.com/assets/9206696/17568926/cb3c8298-5efa-11e6-8567-debe1c210bef.png)
* Change PM2 to Not Met, select BS1 to Supporting and click Save. Expect to see:
![screen shot 2016-08-10 at 1 04 46 pm](https://cloud.githubusercontent.com/assets/9206696/17569003/2b075478-5efb-11e6-9f1a-78fa3fd4c6af.png)

**End of Test**
